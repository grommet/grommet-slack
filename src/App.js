import React, { useState } from 'react';
import {
  Box,
  Button,
  Form,
  FormField,
  Grommet,
  MaskedInput,
  Paragraph,
  grommet,
} from 'grommet';

const apiUrl =
  'https://us-central1-grommet-designer.cloudfunctions.net/invites';
// const apiUrl = "http://localhost:8080/images";

const App = () => {
  const [busy, setBusy] = useState();
  const [message, setMessage] = useState();
  return (
    <Grommet full theme={grommet}>
      <Box fill justify="center" align="center" margin="large">
        <Form
          onSubmit={({ value: { email } }) => {
            setBusy(true);
            const body = JSON.stringify({ email });
            fetch(apiUrl, {
              method: 'POST',
              headers: {
                'Content-Type': 'application/json; charset=UTF-8',
                'Content-Length': body.length,
              },
              body,
            })
              .then(response => {
                if (response.ok) {
                  setMessage('Invited!');
                } else {
                  response.text().then(error => {
                    if (error === 'already_in_team')
                      setMessage('Already a member');
                    else setMessage(error);
                  });
                }
              })
              .then(() => setBusy(false));
          }}
        >
          <Paragraph>
            Enter an email address of someone you'd like to invite to the
            grommet slack channels.
          </Paragraph>
          <FormField label="email" name="email" required>
            <MaskedInput
              name="email"
              mask={[
                { regexp: /^[\w\-_.]+$/, placeholder: 'sample' },
                { fixed: '@' },
                { regexp: /^[\w]+$/, placeholder: 'my' },
                { fixed: '.' },
                { regexp: /^[\w]+$/, placeholder: 'com' },
              ]}
            />
          </FormField>
          <Button type="submit" label="Invite" disabled={busy} />
          {message && <Paragraph>{message}</Paragraph>}
        </Form>
      </Box>
    </Grommet>
  );
};

export default App;
