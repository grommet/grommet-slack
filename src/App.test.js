import React from "react";
import { cleanup, render } from "@testing-library/react";
import App from "./App";

describe("App", () => {
  afterEach(cleanup);

  test("empty", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
