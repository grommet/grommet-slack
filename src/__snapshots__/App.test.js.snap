// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App empty 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <div
    class="StyledBox-sc-13pk1d4-0 ygJUu"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 hNCGjq"
    >
      <img
        aria-label="Gremlin Image"
        class="StyledImage-ey4zx9-0 iBeQsm"
        src="Gremlin.svg"
      />
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 eAyXGY"
    >
      <span
        class="StyledText-sc-1sadyjn-0 DkZvS"
      >
        Sign up for our
      </span>
      <h1
        class="StyledHeading-sc-1rdh4aw-0 hmaxin"
      >
        Grommet Slack Channel
      </h1>
      <p
        class="StyledParagraph-tbetod-0 bvpLHz"
      >
        Join to have access to the Grommet community to stay updated, receive troubleshooting and much more!
      </p>
      <form>
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Email
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
            >
              <input
                aria-label="Email"
                autocomplete="off"
                class="StyledMaskedInput-sc-99vkfa-0 dCkJIz"
                name="email"
                placeholder="sample@email.com"
                value=""
              />
            </div>
          </div>
        </div>
        <button
          class="StyledButton-sc-323bzc-0 kkrMwN"
          type="submit"
        >
          Join Community
        </button>
      </form>
      <div
        class="StyledBox-sc-13pk1d4-0 fJNzIx"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 frzykv"
        >
          Channels
        </h2>
        <p
          class="StyledParagraph-tbetod-0 gHUcCr"
        >
          Popular channels in our Slack community
        </p>
        <div
          class="StyledGrid-sc-1wofa1l-0 gTElmK"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # general
            </span>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # i-made-this
            </span>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # announcements
            </span>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # designer
            </span>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # help
            </span>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 kSwbcK"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fnemch"
            >
              # typescript
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
